src/Core/Domain/Product/Query/SearchProducts.php

```diff
@@ -26,6 +26,7 @@
 
 namespace PrestaShop\PrestaShop\Core\Domain\Product\Query;
 
+use Currency;
```

Comment:
Ok, after seeing the controller code (see below) and since the core provider only returns legacy object... Maybe it's better to use the currency iso code directly here either as a string or using the ValueObject `AlphaIsoCode`

Reply:
:ballot_box_with_check: